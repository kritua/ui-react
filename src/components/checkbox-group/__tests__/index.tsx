import React from 'react';
import { create } from 'react-test-renderer';

import { CheckboxGroup } from '../';

it('No Legend', () => {
    const tree = create(
        <CheckboxGroup
            children="Mock"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Legend', () => {
    const tree = create(
        <CheckboxGroup
            legend="Legend name"
            children="Mock"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Disable', () => {
    const tree = create(
        <CheckboxGroup
            legend="Legend name"
            disabled={true}
            children="Mock"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Direction', () => {
    expect(
        create(
            <CheckboxGroup
                children="Mock"
            />
        ).toJSON()
    ).toMatchSnapshot();

    expect(
        create(
            <CheckboxGroup
                direction="row"
                children="Mock"
            />
        ).toJSON()
    ).toMatchSnapshot();

    expect(
        create(
            <CheckboxGroup
                direction="column"
                children="Mock"
            />
        ).toJSON()
    ).toMatchSnapshot();
});
