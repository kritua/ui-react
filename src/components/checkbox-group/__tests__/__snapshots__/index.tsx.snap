// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Direction 1`] = `
<fieldset
  className="checkbox-group checkbox-group_row"
>
  Mock
</fieldset>
`;

exports[`Direction 2`] = `
<fieldset
  className="checkbox-group checkbox-group_row"
>
  Mock
</fieldset>
`;

exports[`Direction 3`] = `
<fieldset
  className="checkbox-group checkbox-group_column"
>
  Mock
</fieldset>
`;

exports[`Disable 1`] = `
<fieldset
  className="checkbox-group checkbox-group_row"
  disabled={true}
>
  <legend
    className="checkbox-group__legend"
  >
    Legend name
  </legend>
  Mock
</fieldset>
`;

exports[`Legend 1`] = `
<fieldset
  className="checkbox-group checkbox-group_row"
>
  <legend
    className="checkbox-group__legend"
  >
    Legend name
  </legend>
  Mock
</fieldset>
`;

exports[`No Legend 1`] = `
<fieldset
  className="checkbox-group checkbox-group_row"
>
  Mock
</fieldset>
`;
